clvm-derive/src/to_clvm.rs

```diff
@@ -1,53 +1,230 @@
 use proc_macro2::{Ident, Span, TokenStream};
-use quote::quote;
-use syn::{parse_quote, Data, DeriveInput, Fields, GenericParam, Index, TypeParam};
+use quote::{quote, ToTokens};
+use syn::{
+    parse_quote, spanned::Spanned, Data, DeriveInput, Expr, Fields, FieldsNamed, FieldsUnnamed,
+    GenericParam, Index,
+};
 
-use crate::helpers::{add_trait_bounds, parse_args, Repr};
+use crate::{
+    helpers::{add_trait_bounds, parse_clvm_attr, Repr},
+    macros::{repr_macros, Macros},
+};
 
-pub fn to_clvm(mut ast: DeriveInput) -> TokenStream {
-    let args = parse_args(&ast.attrs);
-    let crate_name = quote!(clvm_traits);
+#[derive(Default)]
+struct FieldInfo {
+    field_names: Vec<Ident>,
```

Comment:
as far as I can tell, the only think you use these names for are in destructure expressions. The only important property is that they are unique, right? In theory you could just name them all `field_1`, `field_2`, etc.

Does it improve error messages to use the actual field names (when you have them)?

If not, it might simplify the code somewhat, if you don't have to extract the names, but you can just generate them on the fly where you need them

Reply:
It's used by the initializer for destructuring like you mentioned:
```rs
let initializer = quote!({ #( #field_names, )* });
```

You need to know the names of the parameters to destructure them. It's possible to alias it to `field_1` but there's no need that I can think of. And it should improve error messages not to, yeah

Reply:
Yes, but this is for when there are named fields. For example:

```rust
struct X {
  a: i32,
  b: i32,
}

enum X {
  Y {
    a: i32,
    b: i32,
  }
}
```

You cannot construct `Self` or `Self::Y` in these cases without using the proper names of the fields.

Reply:
When the fields are unnamed, an indexed name is used instead